docs/adr/0008-use-tracing.md

```diff
@@ -0,0 +1,148 @@
+# Replace logging and error reporting infrastructure with tracing.
+
+* Status: proposed
+* Deciders:
+  * AppServices component teams
+    * sync and related components: (decider name, date)
+    * suggest, remote-settings, etc (ie, DISCO): (decider name, date)
+    * (please add your team and decider name here!)
+  * moz-central releng/build teams: ?
+* Date: May 2025
+* Feedback deadline: May 15 2025.
+
+## Context and Problem Statement
+
+Rust code in [application-services](https://github.com/mozilla/application-services/) needs support for diagnostics.
+Specifically, we need support for logging and for explicit error reporting, as described below.
+
+### Logging
+
+Our components must be able to generate logging. These logs must be useful both when a component is being developed
+(eg, sent to the console when running unit tests etc), but also useful when being run in a product
+(eg, sent to the browser and captured normally for that browser - eg, the "console" on desktop, logcat on Android, etc.)
+
+We currently use the `log` crate and wire this up best we can.
+
+### Error Reporting
+
+There's an [error support crate](https://github.com/mozilla/application-services/tree/main/components/support/error),
+designed explicitly for error reporting for the applications. Unlike logins, these error reports are intended to
+be reported to an external service - eg, Android and iOS both use Sentry; Desktop doesn't have this, but it should.
+
+### Problems with the current approaches.
+
+#### Problems using `log`
+
+The main problem with the `log` module is the concept of a [global "max level"](https://docs.rs/cli-log/latest/cli_log/fn.set_max_level.html)
+that's set for all crates.
+Gecko sets the global max level to `Info` - anything more verbose causes [performance regressions in other crates](https://bugzilla.mozilla.org/show_bug.cgi?id=1874215).
+
+In practice, this means there are cases where our browsers are unable to see debug logs from our Rust code.
+
+#### Problems with the error reporter
+
+None identified for mobile - but is not implemented at all for Desktop.
+
+The opportunity is to better align the logging and error-reporting requirements into a single facility while introducing this capability
+to desktop. In the first instance, we will arrange for Javascript code running in Desktop Firefox to be notified on such reports,
```

Comment:
After thinking about this a bit more, I'm not sure this needs to be the first step.  How about removing the "In the first instance..." part and just keeping the "However, a final decision about what it does with them is TBD."